drv/fpga-devices/Cargo.toml

```diff
@@ -6,14 +6,19 @@ edition = "2021"
 [dependencies]
 userlib = {path = "../../sys/userlib"}
 ringbuf = {path = "../../lib/ringbuf"}
+drv-i2c-api = {path = "../../drv/i2c-api", optional = true}
+drv-i2c-devices = {path = "../../drv/i2c-devices", optional = true}
+drv-fpga-api = {path = "../../drv/fpga-api"}
 drv-spi-api = {path = "../../drv/spi-api"}
 drv-stm32xx-sys-api = {path = "../stm32xx-sys-api"}
-drv-fpga-api = {path = "../../drv/fpga-api"}
-num-traits = { version = "0.2.12", default-features = false }
+num-traits = {version = "0.2.12", default-features = false}
 cfg-if = "1"
 bitfield = "0.13"
 zerocopy = "0.6.1"
 
+[features]
+front_io = ["drv-i2c-api", "drv-i2c-devices"]
```

Comment:
This feature doesn't appear to be used; `drv-fpga-devices` builds both `ecp5_spi::Ecp5UsingSpi` and `ecp5_spi_mux_pca9538::Driver` all the time.

I think this is fine, just delete the feature.

Reply:
Ah yes, it needs a `feature` annotation in `lib.rs` to be active. But yes, it's fine to leave this out.